src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -141,6 +377,120 @@ CHIP_ERROR ThermostatAttrAccess::Write(const ConcreteDataAttributePath & aPath,
             return statusIB.ToChipError();
         }
         break;
+    case Presets::Id:
+        {
+            if (presetsSupported == false)
+            {
+                StatusIB statusIB(ToInteractionModelStatus(EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE));
+                return statusIB.ToChipError();                
+            }
+
+            bool currentlyEditing = false;
+            EmberAfStatus status = PresetsEditable::Get(aPath.mEndpointId, &currentlyEditing);
+            if (status != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                StatusIB statusIB(ToInteractionModelStatus(status));
+                return statusIB.ToChipError();
+            }
+            if (currentlyEditing == false)
+            {
+                StatusIB statusIB(ToInteractionModelStatus(EMBER_ZCL_STATUS_INVALID_IN_STATE));
```

Comment:
are we still using EMBER_ZCL ?

Reply:
`ToInteractionModelStatus(EMBER_ZCL_STATUS_INVALID_IN_STATE)` is just `Status::InvalidInState`, no?

So: we do have some EMBER_ZCL around, but it's completely unnecessary here.

Reply:
You'll have to forgive me not know which error codes to use, I'm just looking at other examples in the SDK and trying to do "when in rome".

bool emberAfThermostatClusterSetpointRaiseLowerCallback(app::CommandHandler * commandObj,
                                                        const app::ConcreteCommandPath & commandPath,
                                                        const Commands::SetpointRaiseLower::DecodableType & commandData)

is using this sort of thing, so staying consistent with the file.

Perhaps SDK cleanup should be a different job, and mixing methods in a file should be frowned on :P